eclipse-scout-core/src/table/columns/LookupEditor.ts

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright (c) 2010, 2024 BSI Business Systems Integration AG
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+import {
+  CellEditorPopup, CellEditorRenderedOptions, ChildModelOf, Column, Dimension, Event, HtmlCompPrefSizeOptions, InitModelOf, ListBox, LookupBox, LookupCall, LookupCallOrModel, LookupColumn, LookupResult, Popup, PopupLayout, scout,
+  SmartField, strings, Table, Tree, TreeBox, TreeLayout, ValueField, ValueFieldEventMap, ValueFieldModel, WidgetPopup
+} from '../../index';
+
+export class LookupEditor<TValue> extends ValueField<TValue[]> implements LookupEditorModel<TValue> {
+  declare model: LookupEditorModel<TValue>;
+  declare eventMap: LookupEditorEventMap<TValue>;
+
+  lookupCall: LookupCall<TValue> = null;
+  browseHierarchy = false;
+  browseMaxRowCount = SmartField.DEFAULT_BROWSE_MAX_COUNT;
+
+  protected _popup: WidgetPopup<LookupBox<TValue>> = null;
+
+  protected override _init(model: InitModelOf<this>) {
+    super._init(model);
+
+    this.lookupCall = LookupCall.ensure(this.lookupCall, this.session);
+    this.lookupCall.setMaxRowCount(this.browseMaxRowCount);
+  }
+
+  protected override _render() {
+    super._render();
+    this.$field.addClass('input-field focused');
+  }
+
+  protected override _formatValue(value: TValue[]): string | JQuery.Promise<string> {
+    if (this._popup) {
+      return this._popup.content.displayText;
+    }
+    return LookupColumn.formatValues(value, this.lookupCall, lookupCall => this.trigger('prepareLookupCall', {lookupCall}));
+  }
+
+  protected override _renderDisplayText() {
+    this.$field.html(strings.nl2br(this.displayText || ''));
+    this.invalidateLayoutTree();
+  }
+
+  onCellEditorRendered(options: CellEditorRenderedOptions<TValue[]>) {
+    this._openPopup(options.cellEditorPopup);
+  }
+
+  protected _openPopup(cellEditorPopup: CellEditorPopup<TValue[]>): void {
+    this._popup = scout.create(WidgetPopup<LookupBox<TValue>>, {
+      parent: this,
+      anchor: this,
+      horizontalAlignment: Popup.Alignment.LEFTEDGE,
+      verticalAlignment: Popup.Alignment.BOTTOM,
+      closeOnAnchorMouseDown: false,
+      cssClass: 'lookup-cell-editor-popup',
+      animateResize: false,
+      content: {
+        ...this._createLookupBoxModel(),
+        id: 'LookupBox'
+      }
+    });
+
+    const lookupBox = this._popup.widget('LookupBox', LookupBox<TValue>);
+
+    // sync value
+    lookupBox.on('propertyChange:value', e => this.setValue(e.newValue));
+
+    // pass popup events to editor
+    lookupBox.on('prepareLookupCall', e => this.trigger('prepareLookupCall', {
+      lookupCall: e.lookupCall
+    }));
+
```

Comment:
Remove blank line (belongs to block "pass popup events to editor") and format both event handlers the same way.